### PR TITLE
Updating the node operator guide

### PR DESCRIPTION
**Pull request by vishalchangrani**
This PR updates the node operator to make the Flow node nomenclature closer to Ethereum to allow folks from the Ethereum community to be able to easily understand the different Flow node types. 
- Changes observer to ~~Light Client.~~ Light Node (based on [explanation](https://github.com/onflow/docs/pull/231#issuecomment-1700011757) from Alex)
- Introduces the concept of Full node, which is any Flow staked node.
- Classified SN,LN,EN and VN as validator node (under Full node)
The mental model I had is:

Flownodes diagram

(Flow.com will have additional changes to describe how run a node at a much higher level with a call to action pointing to this page)

### CODE REVIEW

**Comment by AlexHentschel** [comment]
**I would suggest to reference the observer node as "Light _Node_"** (not "light _client_").

To the best of my knowledge, there is no properly defined terminology. However, in the context of Flow, there is an important differentiation between what I call a light node vs a light client:

### Light _Node_ (previously called Observer)
* The light nodes are designed to be continuously online and:
   * observes consensus, by downloading and indexing the blocks (though not the execution state)
   * locally store a continuous history of blocks (the history doesn't need to go back to genesis, but they are generally keeping a longer history of blocks)
   * locally validate that consensus rules are followed
* The important aspect detail is: observers only work if they have a continuous history of blocks. They can be offline, but have to catch up before they can answer queries about recent blocks.
* It can (but doesn't have to) replicate some of the execution state, store it locally, index the replicated part of the state.
* To run a Light _Node_, you need a dedicated computer (maybe only laptop-grade hardware, but at least that depending on how much execution state you want to store locally and how compute-intensive your state queries are) which should be _continuously online_.
* Reasons people might want to run an observer:
   * they care about more than a few accounts
   * they want to execute frequent and complex state queries,  where it is more efficient to replicate some of the execution state (with a limited history) locally, validate its integrity once and then work with the local (already validated) state data
   * low latency of state and event queries 
   * enterprise-level security or reporting requirements (e.g. exchanges, professional investors, governments, etc)

From a software perspective, the Light _Node_ is quite similar to the archive node (store and index blocks, store and index execution state), except that is does not exhaustively replicate the _entire execution state_ and does not commit to keeping a super-long history .

### Light _Client_
* We are architecting light clients to run on embedded hardware (phones, raspberry Pi, smart-home devices and high-end spectrum of micro controllers). These either do not have the hardware resources (in the case of micro controllers), or the user does not want to expend the hardware resources (e.g. phones) to continuously follow the blocks and index them. 
* A light client achieves this by _not_ validating consensus on a continuous basis, but instead receiving consistency proofs that a supermajority of consensus nodes have committed the respective data. 
* Light clients are safe. Just that they won't be able to notice some attacks on the network that observers would be able to detect. So in a sense, Light clients do not really contribute to network safety, observers contribute a bit (they validate consensus), and full nodes contribute a lot (validate consensus and secure the network additionally through their role-specific tasks)
* Light _Clients_ are designed to only store negligible amounts of execution state locally. Essentially all data is fetched on demand (integrity cryptographically validated locally).
* Light _Clients_ are designed for the consumer
  * with a few accounts 
  * largely interested about changes of these accounts
  * does not want to run a client continuously
  * can wait some seconds (maybe a minute when the client was offline for weeks) until most recent state of their accounts has been loaded
  * does not want to keep long history (via their light client)
  * does not want to run dedicated hardware to trustlessly interact with the chain

More details on light clients:
* [Upgrading Flow's Consensus Follower to boost attack resilience and processing speed](https://flow.com/engineering-blogs/upgrading-flows-consensus-follower-to-boost-attack-resilience-and-processing-speed)
* [Flow’s Data Egress: A Long-Term Vision](https://www.notion.so/dapperlabs/Flow-s-Data-Egress-A-Long-Term-Vision-6498c56d4c28424b9a61fb64a1202cc6?pvs=4)

File: docs/building-on-flow/nodes/node-operation/observer-node.mdx [hunk 1]
```diff
@@ -1,25 +1,25 @@
 ---
-title: Observer Node
-sidebar_label: Observer Node
+title: Light Client a.k.a Observer Node
```
Comment (vishalchangrani) [inline docs/building-on-flow/nodes/node-operation/observer-node.mdx:2]: see preview [here](https://docs-git-vishal-nodechanges-onflow.vercel.app/next/building-on-flow/nodes/node-operation/observer-node#docusaurus_skipToContent_fallback)

File: docs/building-on-flow/nodes/node-operation/index.md [hunk 2]
```diff
@@ -20,15 +20,32 @@ Flow multirole architecture makes it more scalable and provides several node typ
 
 The different types of nodes are described [here](./node-roles.mdx). As node operator, you can choose to run any of the different types of node that best fits your needs.
 
-### Observer node
-The observer node is one of the easiest node to spin up and can be run by Dapp developers who need the latest block data available locally e.g. a wallet application that needs to track the latest block ID and height.
+## Light client a.k.a. Observer node
```
Comment (vishalchangrani) [inline docs/building-on-flow/nodes/node-operation/index.md:23]: see a preview of this page [here](https://docs-git-vishal-nodechanges-onflow.vercel.app/next/building-on-flow/nodes/node-operation)

**Comment by vishalchangrani** [comment]
> **I would suggest to reference the observer node as "Light _Node_"** (not "light _client_").
> 
> To the best of my knowledge, there is no properly defined terminology. However, in the context of Flow, there is an important differentiation between what I call a light node vs a light client:
> 
> ### Light _Node_ (previously called Observer)
> * The light nodes are designed to be continuously online and:
>   
>   * observes consensus, by downloading and indexing the blocks (though not the execution state)
>   * locally store a continuous history of blocks (the history doesn't need to go back to genesis, but they are generally keeping a longer history of blocks)
>   * locally validate that consensus rules are followed
> * The important aspect detail is: observers only work if they have a continuous history of blocks. They can be offline, but have to catch up before they can answer queries about recent blocks.
> * It can (but doesn't have to) replicate some of the execution state, store it locally, index the replicated part of the state.
> * To run a Light _Node_, you need a dedicated computer (maybe only laptop-grade hardware, but at least that depending on how much execution state you want to store locally and how compute-intensive your state queries are) which should be _continuously online_.
> * Reasons people might want to run an observer:
>   
>   * they care about more than a few accounts
>   * they want to execute frequent and complex state queries,  where it is more efficient to replicate some of the execution state (with a limited history) locally, validate its integrity once and then work with the local (already validated) state data
>   * low latency of state and event queries
>   * enterprise-level security or reporting requirements (e.g. exchanges, professional investors, governments, etc)
> 
> From a software perspective, the Light _Node_ is quite similar to the archive node (store and index blocks, store and index execution state), except that is does not exhaustively replicate the _entire execution state_ and does not commit to keeping a super-long history .
> 
> ### Light _Client_
> * We are architecting light clients to run on embedded hardware (phones, raspberry Pi, smart-home devices and high-end spectrum of micro controllers). These either do not have the hardware resources (in the case of micro controllers), or the user does not want to expend the hardware resources (e.g. phones) to continuously follow the blocks and index them.
> * A light client achieves this by _not_ validating consensus on a continuous basis, but instead receiving consistency proofs that a supermajority of consensus nodes have committed the respective data.
> * Light clients are safe. Just that they won't be able to notice some attacks on the network that observers would be able to detect. So in a sense, Light clients do not really contribute to network safety, observers contribute a bit (they validate consensus), and full nodes contribute a lot (validate consensus and secure the network additionally through their role-specific tasks)
> * Light _Clients_ are designed to only store negligible amounts of execution state locally. Essentially all data is fetched on demand (integrity cryptographically validated locally).
> * Light _Clients_ are designed for the consumer
>   
>   * with a few accounts
>   * largely interested about changes of these accounts
>   * does not want to run a client continuously
>   * can wait some seconds (maybe a minute when the client was offline for weeks) until most recent state of their accounts has been loaded
>   * does not want to keep long history (via their light client)
>   * does not want to run dedicated hardware to trustlessly interact with the chain
> 
> More details on light clients:
> 
> * [Upgrading Flow's Consensus Follower to boost attack resilience and processing speed](https://flow.com/engineering-blogs/upgrading-flows-consensus-follower-to-boost-attack-resilience-and-processing-speed)
> * [Flow’s Data Egress: A Long-Term Vision](https://www.notion.so/dapperlabs/Flow-s-Data-Egress-A-Long-Term-Vision-6498c56d4c28424b9a61fb64a1202cc6?pvs=4)
> 
> Context:
> 
> * While light clients are not here yet, we are working towards them. So I think we should shape our terminology such that we have a term for them.

❤️ thank you for this explanation - will change `Light Client` to `Light Node`

File: docs/building-on-flow/nodes/node-operation/index.md [hunk 3]
```diff
@@ -20,15 +20,32 @@ Flow multirole architecture makes it more scalable and provides several node typ
...
+The Archive node follows the chain, stores and indexes both protocol and execution state, and allows retrieval of blocks, collections, transactions and events from the genesis of the current spork.
+
+It also allows script execution and other read-only queries that require the execution state to be read. It can answer any queries from past data e.g. “what was the Flow account balance at height X?”, where X is several thousand blocks in the past.
+
+The archive node is currently in beta and will be available as a GA release in H2 2023 (see [here](https://flow.com/post/flow-blockchain-node-operation-archive-node) for more)
 
-### Access node
+## Full node
+
+A Full node is staked node that serves the network by providing data on request (i.e. gets and shares data on the p2p network).
```
Comment (AlexHentschel) [inline docs/building-on-flow/nodes/node-operation/index.md:43]: Reading this, I feel there is a strong focus on full nodes sharing data. While that is true for some nodes (like ENs and in the future VNs) there are also nodes that don't talk to anyone other than staked nodes. 

I would suggest to differentiate full nodes vs light nodes, whether or not they contribute to protocol progress. I think for Collector, Consensus, Execution, and verification nodes that is pretty obvious. Nevertheless, also Access nodes contribute by routing transactions to collectors -- otherwise, the network could not ingest any transaction, which would make it practically useless. Furthermore, ANs help distribute data to the external world.

In comparison, the network could make meaningful progress without any other nodes. The terminology of light node continues to make sense in this context, because the nodes does not have to contribute any resources to network progress. In comparison, Full nodes are designed to _additionally_ provide resources for protocol-internal tasks. 

ANs are a bit of a hybrid between both worlds. Nevertheless, we indent for node operators to at least support the network with some minimal contributions, like routing transactions from other people, very rate-limited script execution, helping to distribute data to the unstaked P2P network. While an operator can run their AN solely for themselves and not provide any services to the community, it is certainly not the intended way of operating an AN. Hence, I feel the classification of an AN as full node is warranted (as you have it in your diagram).


```suggestion
In a nutshell, Full Nodes are staked network participants that drive network progress, e.g. by creating and executing new block. They are the primary contributors to network safety (all of them validate the correctness of the consensus process and secure the network additionally through their role-specific tasks). In comparison, Light Nodes don't contribute to the networks progress. Though, they help to secure the network by also validating the integrity of the consensus process. 
```

Comment (vishalchangrani) [reply docs/building-on-flow/nodes/node-operation/index.md:43]: thank you - committed and reworded a lil.

File: docs/building-on-flow/nodes/node-operation/index.md
Comment (AlexHentschel) [inline docs/building-on-flow/nodes/node-operation/index.md:1]: What so you think about the following revision of line 52, which currently reads:
> An access node is staked but since it does not participate in the core Flow protocol, it does not receive any staking rewards. 

I would suggest to revise this to (edits highlighted in bold):

> An access node is **minimally staked for network security. Nevertheless, the central goal for Access Nodes is to provide functionality to its node operator. In comparison, contributing to protocol progress (e.g. routing transactions to collector clusters, relaying blocks to the unstaked peer-to-peer network, etc.) should only take up a marginal fraction an Access Node's computational resources. Furthermore, Access Node operators can freely rate-limit the amount of resources their Access Node dedicates to supporting the broader ecosystem. Therefore, Access Nodes do** not receive staking rewards.

File: docs/building-on-flow/nodes/node-operation/index.md [hunk 4]
```diff
@@ -37,8 +54,13 @@ To run an access node, see the [Running a staked node](#running-a-staked-node) s
...
 
-### Collection, Consensus, Verification and Execution node
-If you want your node to participate in the nitty-gritty of Flow protocol and help in block or collection creation, transaction execution, result verification or block verification then you should run one of these four node roles.
+## Validator node
+
+If you want your node to participate in the nitty-gritty of Flow protocol and help in block or collection creation, transaction execution, result verification or block verification then you should run one of these four node roles -
+- Collection
+- Consensus
+- Verification
+- Execution
```
Comment (AlexHentschel) [inline docs/building-on-flow/nodes/node-operation/index.md:63]:
```suggestion
You can also be a core participant in running the Flow network and contribute to securing it. Depending on your preference, you could run one or any combination of the following node roles:
- Collection Nodes collaboratively create batches of transactions (in Flow terminology collections).
- Consensus Nodes create blocks, schedule them for asynchronous execution, and commit execution results once they are verified (so called sealing). In addition, they orchestrate the Flow protocol and enforce protocol compliance. 
- Execution Nodes asynchronously execute blocks. They are the power-houses in the protocol, providing the vast computational resources available to Flow transactions. 
- Verification Nodes check the execution results in a distributed manner.
```

File: docs/building-on-flow/nodes/node-operation/observer-node.mdx [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
 
 ![Observer nodes](observer.png)
 
-## Who should run an observer node?
+## Who should run a light client?
 
-The observer node provides an alternative to running an access node. Hence, it is ideal for Dapps that need access to the latest block data available locally e.g. a wallet application that needs to track the latest block ID and height. In addition to that, access node operators who want to scale their access node endpoints geographically can spin up geographically dispersed observer nodes which can talk to their staked access node and to each other.
+The light client provides an alternative to running an access node. Hence, it is ideal for Dapps that need access to the latest block data available locally e.g. a wallet application that needs to track the latest block ID and height. In addition to that, access node operators who want to scale their access node endpoints geographically can spin up geographically dispersed light clients which can talk to their staked access node and to each other.
```
Comment (AlexHentschel) [inline docs/building-on-flow/nodes/node-operation/observer-node.mdx:16]:
```suggestion
The light client provides an alternative to running an access node. Hence, it is ideal for Dapps that need access to the latest block data locally on a machine they control. Examples include a wallet application that needs to track the latest block ID and height. Alternatively, access node operators that want to scale their access node endpoints geographically can spin up light clients in different regions, which can talk to their staked access node and to each other.
```

**Review by AlexHentschel — APPROVED** [review]
few more wording suggestions. Nothing major. Thanks for the update! Nice work 👏

File: docs/building-on-flow/nodes/node-operation/index.md [hunk 6]
```diff
@@ -20,15 +20,32 @@ Flow multirole architecture makes it more scalable and provides several node typ
 
 The different types of nodes are described [here](./node-roles.mdx). As node operator, you can choose to run any of the different types of node that best fits your needs.
 
-### Observer node
-The observer node is one of the easiest node to spin up and can be run by Dapp developers who need the latest block data available locally e.g. a wallet application that needs to track the latest block ID and height.
+## Light client a.k.a. Observer node
+
+The light client is one of the easiest node to spin up and can be run by Dapp developers who need the latest block data available locally e.g. a wallet application that needs to track the latest block ID and height.
```
Comment (peterargue) [inline docs/building-on-flow/nodes/node-operation/index.md:25]:
```suggestion
The light client is one of the easiest nodes to spin up and can be run by Dapp developers who need the latest block data available locally, e.g. a wallet application that needs to track the latest block ID and height.
```

File: docs/building-on-flow/nodes/node-operation/index.md [hunk 7]
```diff
@@ -20,15 +20,32 @@ Flow multirole architecture makes it more scalable and provides several node typ
...
+The Archive node follows the chain, stores and indexes both protocol and execution state, and allows retrieval of blocks, collections, transactions and events from the genesis of the current spork.
+
+It also allows script execution and other read-only queries that require the execution state to be read. It can answer any queries from past data e.g. “what was the Flow account balance at height X?”, where X is several thousand blocks in the past.
+
+The archive node is currently in beta and will be available as a GA release in H2 2023 (see [here](https://flow.com/post/flow-blockchain-node-operation-archive-node) for more)
 
-### Access node
+## Full node
+
+In a nutshell, Full Nodes are staked network participants that drive network progress, e.g. by creating and executing new block. They are the primary contributors to network safety (all of them validate the correctness of the consensus process and secure the network additionally through their role-specific tasks). In comparison, Light Nodes don't contribute to the networks progress. Though, they help to secure the network by also validating the integrity of the consensus process.
```
Comment (peterargue) [inline docs/building-on-flow/nodes/node-operation/index.md:43]:
```suggestion
In a nutshell, Full Nodes are staked network participants that drive network progress, e.g. by creating and executing new blocks. They are the primary contributors to network safety (all of them validate the correctness of the consensus process and secure the network additionally through their role-specific tasks). In comparison, Light Nodes don't contribute to the networks progress. Though, they help to secure the network by also validating the integrity of the consensus process. 
```

File: docs/building-on-flow/nodes/node-operation/index.md [hunk 8]
```diff
@@ -20,15 +20,32 @@ Flow multirole architecture makes it more scalable and provides several node typ
...
+It also allows script execution and other read-only queries that require the execution state to be read. It can answer any queries from past data e.g. “what was the Flow account balance at height X?”, where X is several thousand blocks in the past.
+
+The archive node is currently in beta and will be available as a GA release in H2 2023 (see [here](https://flow.com/post/flow-blockchain-node-operation-archive-node) for more)
 
-### Access node
+## Full node
+
+In a nutshell, Full Nodes are staked network participants that drive network progress, e.g. by creating and executing new block. They are the primary contributors to network safety (all of them validate the correctness of the consensus process and secure the network additionally through their role-specific tasks). In comparison, Light Nodes don't contribute to the networks progress. Though, they help to secure the network by also validating the integrity of the consensus process. 
+- The Access node is a full node that serves as an RPC node and acts as a gateway node for the network.
+- The Validator node (Collection, Consesnus, Verification and Execution) is a full node that plays a role in block generation.
```
Comment (peterargue) [inline docs/building-on-flow/nodes/node-operation/index.md:45]:
```suggestion
- The Validator node (Collection, Consensus, Verification and Execution) is a full node that plays a role in block generation.
```

**Review by peterargue — COMMENTED** [review]
fixed a couple small typos, but otherwise it looks good

File: docs/building-on-flow/nodes/node-operation/index.md
Comment (vishalchangrani) [reply docs/building-on-flow/nodes/node-operation/index.md:1]: made that change

**Comment by vishalchangrani** [comment]
@AlexHentschel @peterargue - Just wanted to call out - I am intentionally avoiding the phrase light node `previously known as observer node` and instead saying `light node a.k.a. observer node` since the word `observer node` is far too prevalent in our documentation, past communications and even in the docker image.
Forcing a rename now will just lead to a lot of confusion.
